garak/resources/loaders/data_sources.py

```diff
@@ -0,0 +1,195 @@
+"""Data loaders used for loading external data:
+(Currently supported)
+- Local files (.txt, .json)
+- HTTP(S) URLs
+
+(Future)
+- HF Datasets
+- MLFlow data
+.... and more
+"""
+
+import json
+import logging
+from pathlib import Path
+from typing import List, Union
+
+
+class DataLoader:
+    """Data loader supporting multiple source types.
+    
+    Examples:
+        # Local file
+        data = DataLoader.load("/path/to/file.json")
+        
+        # HTTP URL
+        data = DataLoader.load("https://example.com/path/to/file.json")
+    """
+    
+    @staticmethod
+    def load(source: Union[str, Path], **kwargs) -> List[str]:
+        """Main method to load data from external sources.
+        
+        Args:
+            source: Path/URI to data source
+                - Local: "file.txt", "file.json"
+                - HTTP(S): "https://example.com/file.txt"
+            **kwargs: Additional parameters passed to specific loaders
+            
+        Returns:
+            List of strings
+            
+        Raises:
+            ValueError: If source type unsupported or loading fails
+        """
+        source_str = str(source)
+        
+        # Route to appropriate loader based on URI scheme
+        if source_str.startswith('http://') or source_str.startswith('https://'):
+            return HTTPDataLoader.load(source_str, **kwargs)
+        else:
+            return LocalFileLoader.load(source_str, **kwargs)
+
+
+class LocalFileLoader:
+    """Loader for local files."""
+    
+    @staticmethod
+    def load(file_path: Union[str, Path], **kwargs) -> List[str]:
+        """Load data from local file.
+        
+        Supports: .txt (one per line), .json (array or object)
+        """
+        file_path = Path(file_path)
```

Comment:
recommend sanitising `file_path` to ensure it's within XDG dirs, locking out path traversal attacks